core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -181,16 +184,42 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
       else consumer.pause(java.util.Collections.singleton(tp))
     }
 
+    def tryPoll(timeout: Long): ConsumerRecords[K, V] =
+      try {
+        val records = consumer.poll(timeout)
+        wakeups = 0
+        records
+      }
+      catch {
+        case w: WakeupException =>
+          wakeups = wakeups + 1
+          if (wakeups == settings.maxWakeups) {
+            log.error("WakeupException limit exceeded, stopping.")
+            context.stop(self)
+          }
+          else {
+            log.warning(s"Consumer interrupted with WakeupException after timeout. Message: ${w.getMessage}. " +
+              s"Current value of akka.kafka.consumer.wakeup-timeout is ${settings.wakeupTimeout}")
+          }
+          null
+        case NonFatal(e) =>
+          log.error(e, "Exception when polling from consumer")
+          context.stop(self)
+          null
+      }
+      finally
+        wakeupTask.cancel()
+
     if (requests.isEmpty) {
       try {
         // no outstanding requests so we don't expect any messages back, but we should anyway
         // drive the KafkaConsumer by polling
 
         def checkNoResult(rawResult: ConsumerRecords[K, V]): Unit =
-          if (!rawResult.isEmpty)
+          if (rawResult != null && !rawResult.isEmpty)
             throw new IllegalStateException(s"Got ${rawResult.count} unexpected messages")
 
-        checkNoResult(consumer.poll(0))
+        checkNoResult(tryPoll(0))
```

Comment:
This will cancel the `wakeupTask`, but it's needed for the below while loop also.
I think you have to cancel outside of `tryPoll`